src/lecturer/components/ListView/ListView.tsx

```diff
@@ -0,0 +1,203 @@
+import { ImportExport } from '../importExport/ImportExport';
+
+
+export interface TitleType {
+    title: string;
+}
+
+interface ListViewProps {
+    getContainer: (() => TitleType[]);
+    setContainer: ((container: TitleType[]) => void);
+    exportFilename: string;
+    createEditPathname: string;
+    listElements: string;
+}
+
+export function ListView(props: ListViewProps) {
+    function isQuestion(object: any): object is Question {
+        return ('title' in object && 'text' in object);
+    }
+
+    interface IndexedElement {
+        index: number;
+        element: TitleType;
+    }
+
+    let getContainer = props.getContainer;
+    let setContainer = props.setContainer;
+    const exportFilename = props.exportFilename;
+    const createEditPathname = props.createEditPathname;
+    const listElements = props.listElements;
+
+    const theme = useTheme();
+    const history = useHistory();
+
+
+    const [filterFn, setFilterFn] = useState({ fn: (items: IndexedElement[]) => { return items } });
+
+    const classes = makeStyles({
+        root: {
+            background: theme.palette.secondary.light,
+            gap: "50px",
+            minHeight: "100vh",
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "center",
+            justifyContent: "center",
+            position: "absolute",
+            width: "100%",
+            top: 0,
+            zIndex: -1,
+            paddingTop: 75,
+            paddingBottom: "10px"
+        },
+        cardWrapper: {
+            position: "relative",
+        },
+        list: {
+            backgroundColor: theme.palette.background.paper,
+            width: "500px",
+            height: "500px",
+            overflow: "auto",
+        },
+        bottomMenu: {
+            width: "500px",
+            display: "flex",
+            alignItems: "center",
+            justifyContent: "space-between",
+            "& > div": {
+                marginLeft: "auto"
+            }
+        },
+        deleteBtn: {
+            color: red[700],
+            flexShrink: 0,
+            width: 55.4,
+        },
+        searchInput: {
+            position: "absolute",
+            top: "42px",
+            right: "25px",
+        },
+    })();
+
+    const handleRemoveElement = (index: number) => {
+        const list = getContainer();
+        list.splice(index, 1);
+        setContainer(list);
+
+    }
+    const selectElement = (index: number) => {
+        history.push({
+            pathname: createEditPathname,
+            state: { index: index }
+        });
+    }
+
+    const handleSearch = (e: ChangeEvent<HTMLInputElement>) => {
+        let { value } = e.target;
+        setFilterFn({
+            fn: (items: IndexedElement[]) => {
+                if (value === "")
+                    return items
+                else
+                    return items.filter(x => x.element.title.toLowerCase().includes(value));
+            }
+        })
```

Comment:
This isn't a big deal, but declaring the whole function every time you change the contents of the searchbar doesn't look efficient. That one variable that changes in the function could be used as a state and used in the filtering function.